table/tables/partition.go

```diff
@@ -207,44 +212,53 @@ func generateHashPartitionExpr(ctx sessionctx.Context, pi *model.PartitionInfo,
 	columns []*expression.Column, names types.NameSlice) (*PartitionExpr, error) {
 	var column *expression.Column
 	// The caller should assure partition info is not nil.
-	partitionPruneExprs := make([]expression.Expression, 0, len(pi.Definitions))
-	var buf bytes.Buffer
 	schema := expression.NewSchema(columns...)
-	for i := 0; i < int(pi.Num); i++ {
-		fmt.Fprintf(&buf, "MOD(ABS(%s),(%d))=%d", pi.Expr, pi.Num, i)
```

Comment:
If we prune hash partition fails, do we fall back to the range partition pruning way? If so, we still can't remove the code here.

Reply:
Currently, if hash prune failed, it will union all partitions.

Reply:
Get it